diesel/src/query_builder/upsert/on_conflict_actions.rs

```diff
@@ -18,11 +18,32 @@ where
     }
 }
 
-impl<DB, T> QueryFragment<DB, T> for DoNothing
+impl<DB> QueryFragment<DB, sql_dialect::on_conflict_clause::PgLikeOnConflictClause> for DoNothing
 where
-    DB: Backend,
-    T: sql_dialect::on_conflict_clause::SupportsOnConflictClause,
+    DB: Backend<
+        OnConflictClause = crate::backend::sql_dialect::on_conflict_clause::PgLikeOnConflictClause,
+    >,
+{
+    fn walk_ast<'b>(&'b self, mut out: AstPass<'_, 'b, DB>) -> QueryResult<()> {
+        out.push_sql(" DO NOTHING");
+        Ok(())
+    }
+}
+
+impl<DB> QueryFragment<DB, sql_dialect::on_conflict_clause::MysqlLikeOnConflictClause>
+    for DoNothing
+where
+    DB: Backend<
+        OnConflictClause = crate::backend::sql_dialect::on_conflict_clause::MysqlLikeOnConflictClause,
+    >,
 {
+    ///
+    /// FIXME: This is super wrong right now and will likely cause a panic; MySQL doesn't support
```

Comment:
I would go for just not supporting `.do_nothing()` on mysql for now. That's possible by just not providing this `QueryFragment` impl. If you choose that way, it would be great to have a compile tests for this behaviour.

Reply:
Yeah definitely. I think I was/am struggling with both:
- `do_nothing`
- `on_conflict_do_nothing`

as per [this comment](https://github.com/diesel-rs/diesel/pull/3177#issuecomment-1128415843). And was just trying to understand what the generics represented and if it was possible to their existing bounds to get access to what was needed (essentially any arbitrary column) or amend the trait bound without breaking too much downstream to do so.

But yeah if I can't figure out one or the other, then maybe easiest just to save it for later. Will keep digging in any case